notes/stv/20230125-red-deploy-01.txt

```diff
@@ -0,0 +1,952 @@
+
+    curl "https://www.duckdns.org/update/${duckname:?}/${ducktoken:?}/${zeppelinip:?}"    
+
+    > OK
+    
+    
+    
+# -----------------------------------------------------
+# Enable HTTPS
+#[root@ansibler]
+
+    /deployments/hadoop-yarn/bin/setup-ssl.sh \
+        "${cloudname:?}" \
+        "${configname:?}" \
+    | tee /tmp/setup-ssl.log
+
+
+# Failed
+
+# Looks like the hostname "zeppelin.gaia-dmp.uk" ended up in the nginx config, where the SSL certificate is under "dmp.gaia.ac.uk"
+
+# Check aglais-statis
+
+cat /opt/aglais/aglais-status.yml 
+
+aglais:
+  status:
+    deployment:
+      type: hadoop-yarn
+      conf: zeppelin-54.86-spark-6.26.43
+      name: iris-gaia-red-20230125
+      date: 20230125T123051
+      hostname: zeppelin.gaia-dmp.uk
+  spec:
+    openstack:
+      cloud:
+        base: arcus
+        name: iris-gaia-red
+
+
+# Manually change to
+
+aglais:
+  status:
+    deployment:
+      type: hadoop-yarn
+      conf: zeppelin-54.86-spark-6.26.43
+      name: iris-gaia-red-20230125
+      date: 20230125T123051
+      hostname: dmp.gaia.ac.uk
+  spec:
+    openstack:
+      cloud:
+        base: arcus
+        name: iris-gaia-red
+
+
+# Rerun nginx / SSL installation
+
+    /deployments/hadoop-yarn/bin/setup-ssl.sh \
+        "${cloudname:?}" \
+        "${configname:?}" \
+    | tee /tmp/setup-ssl.log
+
+
+
+	> 
+	TASK [Create Cronjob to renew certificate] *************************************
+	changed: [zeppelin] => {"changed": true, "envs": [], "jobs": ["Renew Certificate with certbot at 4:10 everyday"]}
+
+	PLAY RECAP *********************************************************************
+	zeppelin                   : ok=7    changed=4    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
+
+	/
+
+
+
+
+# -----------------------------------------------------
+# Check the Nginx service status.
+#[root@ansibler]
+
+    ssh zeppelin \
+        '
+        systemctl status nginx.service
+        '
+        
+        >
+		Jan 25 15:57:36 iris-gaia-red-20230125-zeppelin systemd[1]: Starting The nginx HTTP and reverse proxy server...
+		Jan 25 15:57:36 iris-gaia-red-20230125-zeppelin nginx[82949]: nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
+		Jan 25 15:57:36 iris-gaia-red-20230125-zeppelin nginx[82949]: nginx: configuration file /etc/nginx/nginx.conf test is successful
+		Jan 25 15:57:36 iris-gaia-red-20230125-zeppelin systemd[1]: Started The nginx HTTP and reverse proxy server.
+
+
+
+
+# -----------------------------------------------------
+# Test the HTTPS endpoint using firefox.
+#[user@desktop]
+
+
+    firefox \
+        --new-window \
+        'https://dmp.gaia.ac.uk/'
+
+
+        # Success
+        
+# -----------------------------------------------------
+# Test the HTTP endpoint using firefox.
+#[user@desktop]
+ 
+ 
+     firefox \
+        --new-window \
+        'http://dmp.gaia.ac.uk/'
+
+
+        # Success / Redirected to HTTPS page
+        
```

Comment:
Checking the HTTP to HTTPS redirect visually using Firefox is not enough.  We need an explicit test that verifies the redirect returns the correct header fields using `curl` or Python (#1089).